pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -129,10 +133,23 @@ func (c *cephStatusChecker) updateCephStatus(status *cephclient.CephStatus) erro
 	}
 
 	cephCluster.Status.CephStatus = toCustomResourceStatus(cephCluster.Status, status)
+
 	if err := opcontroller.UpdateStatus(c.client, cephCluster); err != nil {
 		return errors.Wrapf(err, "failed to update cluster %q status", clusterName.Namespace)
 	}
 
+	var condition cephv1.ConditionType
+	var reason, message string
+	condition = cephv1.ConditionReady
+	reason = "ClusterCreated"
+	message = "Cluster created successfully"
+	if c.isExternal {
+		condition = cephv1.ConditionConnected
+		reason = "ClusterConnected"
+		message = "Cluster connected successfully"
+	}
+	config.ConditionExport(c.context, c.clusterInfo.NamespacedName(), condition, v1.ConditionTrue, reason, message)
```

Comment:
The failed condition is being set somewhere, we just weren't ever setting it to connected?

Reply:
We were but in the controller.go which was too early to determine whether the cluster was actually connected or not.